src/SW_netCDF.c

```diff
@@ -0,0 +1,591 @@
+ * @param[in] nDimY Size of dimension "y"
+ * @param[in] domainID ID of the domain variable within the domain netCDF
+ * @param[in] xID ID of the x variable within the domain netCDF
+ * @param[in] yID ID of the y variable within the domain netCDF
+ * @param[in] x_bndsID ID of the x_bnds variable within the domain netCDF
+ * @param[in] y_bndsID ID of the y_bnds variable within the domain netCDF
+*/
+void fill_xy_vars(int nDimX, int nDimY, int domainID,
+        int xID, int yID, int x_bndsID, int y_bndsID) {
+
+    double setValsX[nDimX], setValsY[nDimY];
+    double setValsXBnds[nDimX][NUMBNDS], setValsYBnds[nDimY][NUMBNDS];
+    double Xval = 2000.0, xBndsVal = 0.0, xIncrement = 4000.0;
+    double Yval = 2503826.1, yBndsVal = 2501326.1, yIncrement = 5000.0;
+    float DomVal = 1.0, DomIncrement = 1.0, setValsDom[nDimY][nDimX];
+    int indexX, indexY, indexBnds;
+
+    // NOTE: startXYDom works for both 1D and 2D variables, it seems
+    // the second element of the array is ignored when a 1D variable
+    // is being written to
+    size_t startXYDom[] = {0, 0}, Xcount[] = {nDimX}, Ycount[] = {nDimY},
+        DomCount[2] = {nDimY, nDimX};
+    size_t xBndsCount[] = {nDimX, NUMBNDS}, yBndsCount[] = {nDimY, NUMBNDS};
+
+    // Fill "x" and "x_bnds"
+    // Because "x_bnds" is a 2D variable, use a nested loop to fill the columns
+    // "xBndsVal" is only incremented if it is not the last column
+    for(indexX = 0; indexX < nDimX; indexX++) {
+        setValsX[indexX] = Xval;
+        Xval += xIncrement;
+
+        for(indexBnds = 0; indexBnds < NUMBNDS; indexBnds++) {
+            setValsXBnds[indexX][indexBnds] = xBndsVal;
+
+            if(indexBnds == 0) {
+                xBndsVal += xIncrement;
+            }
+        }
+    }
+
+    nc_put_vara_double(OPEN_NC_ID, xID, startXYDom, Xcount, &setValsX[0]);
+    nc_put_vara_double(OPEN_NC_ID, x_bndsID, startXYDom, xBndsCount,
+                       &setValsXBnds[0][0]);
+
+    // Fill "y" and "y_bnds"
+    // Because "y_bnds" is a 2D variable, use a nested loop to fill the columns
+    // "yBndsVal" is only incremented if it is not the last column
+    for(indexY = 0; indexY < nDimY; indexY++) {
+        setValsY[indexY] = Yval;
+        Yval += yIncrement;
+
+        for(indexBnds = 0; indexBnds < NUMBNDS; indexBnds++) {
+            setValsYBnds[indexY][indexBnds] = yBndsVal;
+
+            if(indexBnds == 0) {
+                yBndsVal += yIncrement;
+            }
+        }
+    }
+
+    nc_put_vara_double(OPEN_NC_ID, yID, startXYDom, Ycount, &setValsY[0]);
+    nc_put_vara_double(OPEN_NC_ID, y_bndsID, startXYDom, yBndsCount,
+                       &setValsYBnds[0][0]);
+
+    // Fill "domain"
+    for(indexY = 0; indexY < nDimY; indexY++) {
+        for(indexX = 0; indexX < nDimX; indexX++) {
+            setValsDom[indexY][indexX] = DomVal;
+            DomVal += DomIncrement;
+        }
+    }
+
+    nc_put_vara_float(OPEN_NC_ID, domainID, startXYDom, DomCount,
```

Comment:
why are `suids` written as float and not as one of the (long) integer types?

Reply:
I am going off the example domain netCDFs, and since they are floats in the examples, they are being generated as floats.